src/soter/openssl/soter_sign_ecdsa.c

```diff
@@ -89,18 +89,25 @@ soter_status_t soter_sign_init_ecdsa_none_pkcs8(soter_sign_ctx_t* ctx,
     return err;
 }
 
-soter_status_t soter_sign_export_key_ecdsa_none_pkcs8(soter_sign_ctx_t* ctx,
-                                                      void* key,
-                                                      size_t* key_length,
-                                                      bool isprivate)
+soter_status_t soter_sign_export_private_key_ecdsa_none_pkcs8(const soter_sign_ctx_t* ctx,
+                                                              void* key,
+                                                              size_t* key_length)
 {
-    if (!ctx || !ctx->pkey) {
+    if (!ctx) {
```

Comment:
why we remove a check for key?

Reply:
> why we remove a check for key?

`ctx->pkey` is not immediately referenced in this function. Instead `soter_ec_export_{private,public}_key()` is responsible for checking, since it's the one using it.

In practice, those calls are going to be inlined, so there is no performance loss here from moving the check down the call stack, closer to its use.

Reply:
i see thank you